central/platform/reprocessor/reprocessor_impl_test.go

```diff
@@ -85,7 +95,16 @@ func (s *platformReprocessorImplTestSuite) TestStartAndStop() {
 	// Mock calls made by alert reprocessing loop
 	proceedAlertLoop := concurrency.NewSignal()
 	inAlertLoop := concurrency.NewSignal()
-	s.alertDatastore.EXPECT().GetByQuery(gomock.Any(), gomock.Any()).Return(testAlerts(), nil).Times(1)
+	s.alertDatastore.EXPECT().WalkByQuery(gomock.Any(), gomock.Any(), gomock.Any()).
+		DoAndReturn(func(ctx context.Context, query *v1.Query, fn func(*storage.Alert) error) error {
+			for _, a := range testAlerts() {
+				err := fn(a)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		}).Times(1)
```

Comment:
nit:
The pattern appears in multiple locations in this file, would it make sense to extract it in a helper function ?
For the sake of symmetry, it could also make sense to extract a function for expected `WalkByQuery` that does not process anything.

Reply:
Done